### ISSUE TITLE
Instructions for getting from Xcode to iPhone

### ISSUE THREAD

**Issue by sdaitzman** [thomasfinch/GammaThingy#18]
I've downloaded the project but I'm not familiar with iOS development. Can the readme have a brief set of instructions that detail how you install this?

**Comment by tschoof** [comment]
So I got it to work without a paid developer account. I'm gonna try a quick and dirty rundown, I'm not sure if I can remember all the right steps though.
* The easiest way to begin should be to clone this repo via the link on the main page to the Github desktop app
* then, install and launch XCode, open a new project, an iOS application
* for the product name, enter "de.nautigg.me.thomasfinch.GammaTest" (it'll make sense), think of an organization identifier and then create that project
* in the "General" tab, add your Apple ID to the "Team" thingy
* then, "Fix Issue", and the certificate or whatever is created
* now everything should be ready to leave and close this project, and launch the cloned GammaTest project
* under "Build settings" and "Code signing", there should be an "iPhone developer" identity to select in all the fields, and a profile for "provisioning profile" as well
* back in the "General" tab, you need to make the Bundle Identifier com.YourTeamName.de.nautigg.me.thomasfinch.GammaTest
* in the thing next to the play button, select your usb-connected device, and go. it'll build with some errors
* you should be able to see the app on your phone
* now you just need to allow your dev profile on iOS in General, Profiles to run apps.

That was it for me, hope that helps for now

**Comment by galambalazs** [comment]
https://www.reddit.com/r/apple/comments/3o5sg7/tutorial_how_to_get_flux_on_ios_9/

Video: https://www.youtube.com/watch?v=D_01P4-ubdA

**Comment by joshuaswilcox** [comment]
I keep getting
    An App ID with Identifier 'de.nautigg.me.thomasfinch.GammaTest' is not available. Please enter a different string.

**Comment by dso371** [comment]
I simply cannot get this to work. I've followed all the instructions as far as I can tell, and I end up with an app icon that is just an iOS grid without the gamma picture, and when I tap it I get a blank page that says Master in the centre top, Edit in top left, and + in top right, and if I hit + I just get a time stamp! What have I done wrong? Thanks, I'd love to get this to work on my phone.

**Comment by ANGOmarcello** [comment]
And you made sure to checkout this adress in the Xcode menu? https://github.com/thomasfinch/GammaThingy.git

**Comment by dso371** [comment]
Thanks, I got it to work after trying again with the instructions at the top and using checkout as you suggest, with that address. Not sure what I did wrong before but now it's installed fine, thank you.

But here's the irony - I thought the app would allow me to *decrease* orangeness but it only allows me to *increase* it! The lowest setting on the app is actually identical to not enabling it at all! That's a pity, since I like increased white point and cooler screens. Any chance of enabling this in a future version? Many thanks again, David

**Comment by ANGOmarcello** [comment]
Maybe we should include the idownloadblog link in the compiling part of the readme @thomasfinch

**Comment by dso371** [comment]
Yes that would be a good idea but you need to tell people to pay attention both to the idownload instructions and the ones at the top of this thread.

**Comment by galambalazs** [comment]
added the link to the README in this PR: https://github.com/thomasfinch/GammaThingy/pull/42

**Comment by ANGOmarcello** [comment]
@galambalazs :+1:

**Comment by dpelletier360** [comment]
Ios 9.2 the profile won't show up in the settings app?

**Comment by murr4y** [comment]
Same here, no profile to select with 9.2.

**Comment by rolyreyes** [comment]
It's an iOS bug. Downgrade or wait for a fix.

**Comment by rex** [comment]
Yeah, I've spent about 6 hours trying to get this onto my iPhone (6+, iOS 9.2) to no avail. I've created provisioning profiles, identities, everything I can think of and it still won't build without failing due to code signing issues. Is it because it's 9.2 and Xcode only has the release target of 9.1? I don't think it would be...

**Comment by Caesar113** [comment]
Are you using XCode 7.2 beta? If not, you need to be.

**Comment by dpelletier360** [comment]
I'm using XCode 7.2 beta and I'm on iOS 9.2. I've looked it up and lots of people are also having this issue. As rolyreyes said either wait for a patch or downgrade. I'm personally gonna wait for a patch.

**Comment by rex** [comment]
Sounds like a plan to me. Patch it is.

**Comment by dpelletier360** [comment]
9.2 beta 3 was released today and has resolved the issue of certificates not showing up. If you have the app sideloaded then just go to settings, general, profiles, click on your profile and approve it.

**Comment by rex** [comment]
I just realized I have Xcode stable installed. Not looking forward to uninstalling and reinstalling the latest beta. Most likely will just wait for the stable release channel to reflect the updates. (Unless there's a way to run beta and stable side-by-side?)

**Comment by Caesar113** [comment]
Yes, you can run stable and beta side by side. The apps have different names. Just download the beta from apple.com and drag/drop it into Applications. You may want to check this out as well, because it actually works, unlike this project: https://justgetflux.com/sideload/

**Comment by rex** [comment]
@Caesar113 Holy wow.... That looks amazing. You are a gentleman and a scholar for posting that link. Doesn't an *official* sideload of f.lux kind of make this repo obsolete?

**Comment by thomasfinch** [comment]
It does, as is now noted in the readme. This project was kind of a temporary filler until an official sideload version of f.lux came out, so I’m now considering it finished. The official version of f.lux has much nicer color calculation as well as more polish, so it’s definitely better for users. Of course this project isn’t going anywhere, the code will still be available to read and learn from, but as far as I’m concerned it’s not worth working on anymore.

**Comment by rex** [comment]
@Caesar113 Yeah, it fired right up!

@thomasfinch I agree. But regardless of whether I agree or not, it goes without saying that you have made a tremendous effort with this project to bring a sorely needed tool to f.lux iOS users. Your work here is appreciated so much by so many. 

Now go crack open a cold beer, relax, and let the f.lux developers deal with it :-P :beer:

**Comment by thomasfinch** [comment]
Wow I might have spoken too soon https://justgetflux.com/sideload/?utm_content=bufferaa94f&utm_medium=social&utm_source=twitter.com&utm_campaign=buffer

**Comment by rex** [comment]
@thomasfinch Wow, what the hell? I wonder a few things...

1) Why? It seems seriously heavy handed for Apple to do this for seemingly no reason. The f.lux project has been in place for a long time and is a legitimate project through and through.
2) How do they intend to regulate this? Seems to me that as long as you don't update the OS or Xcode there's nothing they can do in the meantime.

:-/

**Comment by rolyreyes** [comment]
As much as I love flux and continue to use it, running unsigned and unvetted binaries create too many attack vectors for malicious actors.

The flux team does not appear to be willing to share their source regardless of license type, unfortunately. I hope GammaThingy continues to be worked on. 

Thanks for all your work thus far.

**Comment by galambalazs** [comment]
f.lux calculations are arguably better, but in terms of waking up the phone and scheduling change GammaThingy works better for me so far.
(also f.lux spams your phone with notifications which is an ugly workaround)
